model/qti/ImportService.php

```diff
@@ -501,6 +531,7 @@ public function importQtiItem(
                             ) . $validationReport->getMessage()
                         );
                         \common_Logger::i('Item metadata is not valid: ' . $validationReport->getMessage());
+                        $this->checkImportLockTime($startImportTime, $qtiItemResource->getIdentifier());
```

Comment:
I think it would be better to add `final` block and release lock / checkImportLockTime there to avoid duplicated and be safe in case we miss any exception or there will be another return statement

Reply:
Thank you, @SashaTAO, I thought about that. It's a complex function and requires complex refactoring. I guess we need to create new task for that. If I add final with only these 2 actions it will make this method more messy and unclear.